consensus/service/src/tx_manager.rs

```diff
@@ -351,43 +363,213 @@ impl<E: ConsensusEnclaveProxy, L: Ledger, UI: UntrustedInterfaces> TxManager<E,
         Ok(self.enclave.txs_for_peer(&encrypted_txs?, aad, peer)?)
     }
 
-    pub fn get_encrypted_tx_by_hash(&self, tx_hash: &TxHash) -> Option<WellFormedEncryptedTx> {
-        self.lock_cache()
+    /// Get the encrypted transaction corresponding to the given hash.
+    fn get_encrypted_tx_by_hash(&self, tx_hash: &TxHash) -> Option<WellFormedEncryptedTx> {
+        self.cache
             .get(tx_hash)
             .map(|entry| entry.encrypted_tx().clone())
     }
 
-    pub fn num_entries(&self) -> usize {
-        self.lock_cache().len()
-    }
-
-    fn lock_cache(&self) -> MutexGuard<HashMap<TxHash, CacheEntry>> {
-        self.cache.lock().expect("lock poisoned")
+    /// The number of cached entries.
+    fn num_entries(&self) -> usize {
+        self.cache.len()
     }
 }
 
 #[cfg(test)]
-mod tests {
+mod tx_manager_tests {
     use super::*;
     use crate::validators::DefaultTxManagerUntrustedInterfaces;
-    use mc_common::logger::test_with_logger;
+    use mc_attest_core::{IasNonce, Quote, QuoteNonce, Report, TargetInfo, VerificationReport};
+    use mc_attest_enclave_api::{
+        ClientAuthRequest, ClientAuthResponse, ClientSession, EnclaveMessage, PeerAuthRequest,
+        PeerAuthResponse, PeerSession,
+    };
+    use mc_common::{logger::test_with_logger, ResponderId};
+    use mc_consensus_enclave::{Error as EnclaveError, LocallyEncryptedTx, SealedBlockSigningKey};
     use mc_consensus_enclave_mock::ConsensusServiceMockEnclave;
+    use mc_crypto_keys::{Ed25519Public, X25519Public};
+    use mc_ledger_db::Ledger;
+    use mc_sgx_report_cache_api::{Error as SgxReportError, ReportableEnclave};
     use mc_transaction_core_test_utils::{
         create_ledger, create_transaction, initialize_ledger, AccountKey,
     };
     use rand::{rngs::StdRng, SeedableRng};
 
+    // ConsensusEnclave inherits from ReportableEnclave, so the traits have o be re-typed here.
```

Comment:
This isn't ideal, but I think it's better than how we currently declare a mock struct and actually implement these traits: https://github.com/mobilecoinofficial/mobilecoin/blob/master/consensus/enclave/mock/src/lib.rs#L30